rllib/core/rl_trainer/tf/tf_rl_trainer.py

```diff
@@ -18,12 +18,81 @@
 
 
 class TfRLTrainer(RLTrainer):
-    """Base class for RLlib TF algorithm trainers."""
+    """Base class for RLlib TF algorithm trainers.
+
+    Args:
+        module_class: The (MA)RLModule class to use.
+        module_kwargs: The kwargs for the (MA)RLModule.
+        optimizer_class: The optimizer class to use.
+        optimizer_kwargs: The kwargs for the optimizer.
+        scaling_config: A mapping that holds the world size and rank of this
+            trainer. Note this is only used for distributed training.
+        distributed: Whether this trainer is distributed or not.
+        use_tf_function: Whether to trace the update function with tf.function.
+
+    Abstract Methods:
+        compute_gradients: Compute gradients for the module being optimized.
+        apply_gradients: Apply gradients to the module being optimized with respect to
+            a loss that is computed by the optimizer.
+
+    Example:
+        .. code-block:: python
+
+        trainer = MyRLTrainer(module_class, module_kwargs, optimizer_class,
+                optimizer_kwargs, scaling_config)
+        trainer.init_trainer()
+        batch = ...
+        results = trainer.update(batch)
+
+        # add a new module, perhaps for league based training or lru caching
+        trainer.add_module("new_player", NewPlayerCls, new_player_kwargs,
+            NewPlayerOptimCls, new_player_optim_kwargs)
+
+        batch = ...
+        results = trainer.update(batch)  # will train previous and new modules.
+
+        # remove a module
+        trainer.remove_module("new_player")
+
+        batch = ...
+        results = trainer.update(batch)  # will train previous modules only.
+
+        # get the state of the trainer
+        state = trainer.get_state()
+
+        # set the state of the trainer
+        trainer.set_state(state)
+
+    """
+
+    def __init__(
+        self,
+        module_class: Union[Type[RLModule], Type[MultiAgentRLModule]],
+        module_kwargs: Mapping[str, Any],
+        optimizer_class: Union[Type[RLOptimizer], Type[MultiAgentRLOptimizer]],
+        optimizer_kwargs: Mapping[str, Any],
+        scaling_config: Mapping[str, Any],
+        distributed: bool = False,
+        use_tf_function: bool = True,
```

Comment:
I added this flag over here to be able to disable tf tracing, and this meant that I needed to add a docstring as well.